subsys/settings/src/settings.c

```diff
@@ -234,39 +242,80 @@ int settings_commit(void)
 	return settings_commit_subtree(NULL);
 }
 
+static bool skip_commit(int handler_cprio, int cprio, int *next_cprio)
+{
+	if (handler_cprio < cprio) {
+		return true;
+	}
+
+	if (handler_cprio == cprio) {
+		return false;
+	}
+
+	if (*next_cprio == cprio) {
+		*next_cprio = handler_cprio;
+		return true;
+	}
+
+	*next_cprio = MIN(*next_cprio, handler_cprio);
+	return true;
+}
+
 int settings_commit_subtree(const char *subtree)
 {
 	int rc;
 	int rc2;
+	int cprio = INT_MIN;
 
 	rc = 0;
 
-	STRUCT_SECTION_FOREACH(settings_handler_static, ch) {
-		if (subtree && !settings_name_steq(ch->name, subtree, NULL)) {
-			continue;
-		}
-		if (ch->h_commit) {
-			rc2 = ch->h_commit();
-			if (!rc) {
-				rc = rc2;
+	while (true) {
+		int next_cprio = cprio;
```

Comment:
```suggestion
		int next_cprio = cprio + 1;
```
The next priority will always be larger than `cprio`, right?

Reply:
Yes, but it is unknown if there will be any handlers with higher prio value, if there are none the next_prio will remain set to cprio.

Reply:
Hmm, yeah, and that's what you use to to break the loop, right?

Reply:
I think this is an OK design, but needs a few comments for readers as it is non-obvious (and semantically odd that `next = current`). If you state the intent and reasoning for this, it's OK with me

Reply:
Also please move this line to L267, because this repeat with L317 `cprio = next_cprio;`

Reply:
I don't think this is a repeat.